File: uiscrollview/uiscrollview/ViewController.m
//
//  ViewController.m
//  uiscrollview
//
//  Created by admin on 16/7/31.
//  Copyright © 2016年 yicheng. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()<UIScrollViewDelegate>

@property(nonatomic, strong)UIScrollView*myView;
@property(nonatomic,strong)UIImageView*img;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    UIScrollView*view1 = [[UIScrollView alloc]initWithFrame:self.view.bounds];
    view1.backgroundColor = [UIColor yellowColor];
    
    //创建图片
    UIImageView *img = [[UIImageView alloc]initWithFrame:CGRectMake(0, 0, 4457, 888)];
    img.image = [UIImage imageNamed:@"shanghai.jpg"];
    
    self.img = img;
    
    [view1 addSubview:img];
    [self.view addSubview:view1];

    //可滚动的范围
    view1.contentSize = CGSizeMake(4457, 888);
    
    //显示的起始偏移
    view1.contentOffset = CGPointMake(2200, 0);
    
    //滚动条显示/消失
    view1.showsVerticalScrollIndicator = false;
    view1.showsHorizontalScrollIndicator = false;
    
    //内边距
    view1.contentInset = UIEdgeInsetsMake(10, 20, 30, 40);
    
    //弹簧效果
    view1.bounces = false;
    
    //代理/事件
    view1.delegate = self;
    
    //通过手势缩放图片
    //1. 设置缩放范围
    view1.minimumZoomScale = 0.1;
    view1.maximumZoomScale = 5;
    //2.
    
    self.myView = view1;
    
}

//设置要缩放的视图
-(UIView*)viewForZoomingInScrollView:(UIScrollView *)scrollView
{
    return self.img;
}

-(void)scrollViewWillBeginDragging:(UIScrollView *)scrollView
{
    NSLog(@"开始拖拽 %@", NSStringFromCGPoint(self.myView.contentOffset));
}

-(void)scrollViewDidEndDragging:(UIScrollView *)scrollView willDecelerate:(BOOL)decelerate
{
    NSLog(@"结束拖拽 %@", NSStringFromCGPoint(self.myView.contentOffset));
}

-(void)scrollViewDidScroll:(UIScrollView *)scrollView
{
    NSLog(@"拖拽中 %@", NSStringFromCGPoint(self.myView.contentOffset));
}

-(void)scrollViewWillBeginDecelerating:(UIScrollView *)scrollView
{
    NSLog(@"开始减速 %@", NSStringFromCGPoint(self.myView.contentOffset));
}

-(void)scrollViewDidEndDecelerating:(UIScrollView *)scrollView
{
    NSLog(@"减速结束 %@", NSStringFromCGPoint(self.myView.contentOffset));
}


@end

File: scrollView/scrollView/ViewController.swift
//
//  ViewController.swift
//  scrollView
//
//  Created by admin on 16/6/29.
//  Copyright © 2016年 yicheng. All rights reserved.
//

import UIKit

class ViewController: UIViewController, UIScrollViewDelegate {

    @IBOutlet weak var galleryScrollView: UIScrollView!
    @IBOutlet weak var galleryPageControl: UIPageControl!
    
    var timer: NSTimer?
    
    override func viewDidLoad() {
        super.viewDidLoad()
    
        pictureGallery()
    }
    
    func pictureGallery(){   //实现图片滚动播放；
        //image width
        let imageW:CGFloat = self.galleryScrollView.frame.size.width;//获取ScrollView的宽作为图片的宽；
        let imageH:CGFloat = self.galleryScrollView.frame.size.height;//获取ScrollView的高作为图片的高；
        let imageY:CGFloat = 0;//图片的Y坐标就在ScrollView的顶端；
        let totalCount:NSInteger = 5;//轮播的图片数量；
        for index in 0..<totalCount{
            let imageView:UIImageView = UIImageView();
            let imageX:CGFloat = CGFloat(index) * imageW;
            imageView.frame = CGRectMake(imageX, imageY, imageW, imageH);//设置图片的大小，注意Image和ScrollView的关系，其实几张图片是按顺序从左向右依次放置在ScrollView中的，但是ScrollView在界面中显示的只是一张图片的大小，效果类似与画廊；
            imageView.image = UIImage(named: "\(index+1)");
            self.galleryScrollView.showsHorizontalScrollIndicator = false;//不设置水平滚动条；
            imageView.clipsToBounds = true
            imageView.userInteractionEnabled = true
            self.galleryScrollView.addSubview(imageView);//把图片加入到ScrollView中去，实现轮播的效果；
        }
        
        //需要非常注意的是：ScrollView控件一定要设置contentSize;包括长和宽；
        let contentW:CGFloat = imageW * CGFloat(totalCount);//这里的宽度就是所有的图片宽度之和；
        self.galleryScrollView.contentSize = CGSizeMake(contentW, 0)
        self.galleryScrollView.pagingEnabled = true;
        self.galleryScrollView.delegate = self;
        self.galleryPageControl.numberOfPages = totalCount;//下面的页码提示器；
        self.addTimer()
    }
    
    func nextImage(sender:AnyObject!){//图片轮播；
        var page:Int = self.galleryPageControl.currentPage;
        if(page == 4){   //循环；
            page = 0;
        }else{
            page+=1;
        }
        let x:CGFloat = CGFloat(page) * self.galleryScrollView.frame.size.width;
        self.galleryScrollView.contentOffset = CGPointMake(x, 0);//注意：contentOffset就是设置ScrollView的偏移；
    }
    
    //UIScrollViewDelegate中重写的方法；
    //处理所有ScrollView的滚动之后的事件，注意 不是执行滚动的事件；
    func scrollViewDidScroll(scrollView: UIScrollView) {
        //这里的代码是在ScrollView滚动后执行的操作，并不是执行ScrollView的代码；
        //这里只是为了设置下面的页码提示器；该操作是在图片滚动之后操作的；
        let scrollviewW:CGFloat = galleryScrollView.frame.size.width;
        let x:CGFloat = galleryScrollView.contentOffset.x;
        let page:Int = (Int)((x + scrollviewW / 2) / scrollviewW);
        self.galleryPageControl.currentPage = page;
    }
    
    func addTimer(){   //图片轮播的定时器；
        self.timer = NSTimer.scheduledTimerWithTimeInterval(3, target: self, selector: #selector(ViewController.nextImage(_:)), userInfo: nil, repeats: true);
    }


}

